clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1163,10 +1169,21 @@ private Record nextFetchedRecord() {
 
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             // Error when fetching the next record before deserialization.
-            if (corruptLastRecord)
-                throw new KafkaException("Received exception when fetching the next record from " + partition
-                                             + ". If needed, please seek past the record to "
-                                             + "continue consumption.", cachedRecordException);
+            if (corruptLastRecord) {
+                long corruptOffset;
+
+                if (cachedRecordException instanceof InoperativeRecordException)
+                    corruptOffset = ((InoperativeRecordException) cachedRecordException).offset();
+                else if (cachedRecordException instanceof RecordDeserializationException)
+                    corruptOffset = ((RecordDeserializationException) cachedRecordException).offset();
+                else
+                    corruptOffset = nextFetchOffset;
```

Comment:
Not absolutely sure if setting it to the `nextFetchOffset` is correct, but I figured it's the only thing to do.
We enter this `if (corruptLastRecord)` block if `Fetcher#nextFetchedRecord()` throws